src/lib_ccx/ccx_encoders_common.c

```diff
@@ -924,7 +924,8 @@ void dinit_encoder(struct encoder_ctx **arg, LLONG current_fts)
 		write_subtitle_file_footer(ctx, ctx->out + i);
 	}
 
-	free_encoder_context(ctx->prev);
+	if(!ccx_options.hardsubx) // Don't free ctx->prev in case of hardsubx to avoid segfault
+		free_encoder_context(ctx->prev);
```

Comment:
Why is that call to free_encoder_context segfaulting? That's what we should fix, not just not do the call because it segfaults... is it because ctx->prev is null, because it ends in a double-free, or what?

This seems like hiding the problem, not fixing it.

Reply:
@cfsmp3 Yes, you are right. I didn't go deep enough to investigate the cause of the segfault before your comment.

So the prev field of an encoder_ctx is used to process dvb subtitles. They are not at all used in the code that involves processing hard subtitles. And when an encoder_ctx is initialized in init_encoder function, the prev field is not set to NULL and hence, it gets some random value(address). So, ctx->prev != NULL and free_encoder_context tries to free the fields of ctx->prev, which do not exist, and hence a segfault. So this fix is wrong and the actual fix is very simple. Please check my latest fix.